src/components/modals/index.ts

```diff
@@ -0,0 +1,10 @@
+import Modal from "./modal";
+
+export interface ModalProps {
+    isOpen: boolean;
+    title:string;
+    onToggle: React.Dispatch<React.SetStateAction<boolean>>;
```

Comment:
I don't think is is the most elegant approach. From the name of the function I would except that it triggers something after a toggle happens(But that is not the case, its just a dispatch function allowing you to call the parent function).